src/daft-core/src/array/ops/hll_sketch.rs

```diff
@@ -0,0 +1,69 @@
+use crate::{
+    array::ops::as_arrow::AsArrow,
+    datatypes::{FixedSizeBinaryArray, UInt64Array},
+    DataType,
+};
+use hyperloglog::{HyperLogLog, NUM_REGISTERS};
+
+use common_error::DaftResult;
+
+use crate::array::ops::{DaftHllSketchAggable, GroupIndices};
+
+pub const HLL_SKETCH_DTYPE: DataType = DataType::FixedSizeBinary(NUM_REGISTERS);
+
+impl DaftHllSketchAggable for UInt64Array {
+    type Output = DaftResult<FixedSizeBinaryArray>;
+
+    fn hll_sketch(&self) -> Self::Output {
+        let mut hll = HyperLogLog::default();
+        match self.validity() {
+            Some(validity) => {
+                for (index, &value) in self.as_arrow().values_iter().enumerate() {
```

Comment:
favor `.iter()` if you are also calling `get_bit`

Reply:
Updated!